shotgun_api3/shotgun.py

```diff
@@ -4470,6 +4470,28 @@ def _translate_filters_simple(sg_filter):
     if len(values) == 1 and isinstance(values[0], (list, tuple)):
         values = values[0]
 
+    # Payload optimization: Do not send a full object
+    # just send the `type` and `id` when combining related queries
+    disable_entity_optimization = os.environ.get(
+        "SHOTGUN_API_DISABLE_ENTITY_OPTIMIZATION"
+    )
+    if (
+        not disable_entity_optimization
```

Comment:
One problem with this logic, this will match if `SHOTGUN_API_DISABLE_ENTITY_OPTIMIZATION=1`

There is no behavior difference between values 0 and 1 which seems pretty bad.

I'd prefer to convert the value from string to bool 
```python
disable_entity_optimization = os.environ.get(
  "SHOTGUN_API_DISABLE_ENTITY_OPTIMIZATION", ""
).strip().lower() not in ["0", "no", "false"]
```

Reply:
Thanks. I was aware of that and I this it's a valid option for testing purposes.